### PR TITLE
chore: update DETR model card

### PR DESCRIPTION
**Pull request by arpon-kapuria**
# What does this PR do?

This PR updates the model card for DETR, following the template outlined in the issue.

## Before submitting
- [x] This PR improves the docs.

## Who can review?
@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/detr.md [hunk 1]
```diff
@@ -14,135 +14,132 @@ rendered properly in your Markdown viewer.
...
+
 # DETR
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[DETR](https://huggingface.co/papers/2005.12872) consists of a convolutional backbone followed by an encoder-decoder Transformer which can be trained end-to-end for object detection. It greatly simplifies a lot of the complexity of models like Faster-R-CNN and Mask-R-CNN, which use things like region proposals, non-maximum suppression procedure and anchor generation. Moreover, DETR can also be naturally extended to perform panoptic segmentation, by simply adding a mask head on top of the decoder outputs.
+
+> [!TIP]
+> This model was contributed by [nielsr](https://huggingface.co/nielsr). The original code can be found [here](https://github.com/facebookresearch/detr).
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md]:
```suggestion
> This model was contributed by [nielsr](https://huggingface.co/nielsr).
```

File: docs/source/en/model_doc/detr.md [hunk 2]
```diff
@@ -14,135 +14,132 @@ rendered properly in your Markdown viewer.
...
+	</div>
+</div>
+
 # DETR
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[DETR](https://huggingface.co/papers/2005.12872) consists of a convolutional backbone followed by an encoder-decoder Transformer which can be trained end-to-end for object detection. It greatly simplifies a lot of the complexity of models like Faster-R-CNN and Mask-R-CNN, which use things like region proposals, non-maximum suppression procedure and anchor generation. Moreover, DETR can also be naturally extended to perform panoptic segmentation, by simply adding a mask head on top of the decoder outputs.
+
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md:26]:
```suggestion
You can find all the original DETR checkpoints under the [AI at Meta](https://huggingface.co/facebook/models?search=detr) organization.
```

File: docs/source/en/model_doc/detr.md [hunk 3]
```diff
@@ -14,135 +14,132 @@ rendered properly in your Markdown viewer.
...
+> [!TIP]
+> This model was contributed by [nielsr](https://huggingface.co/nielsr). The original code can be found [here](https://github.com/facebookresearch/detr).
+>
+> Click on the DETR models in the right sidebar for more examples of how to apply DETR to different object detection and segmentation tasks.
+
+The example below demonstrates how to perform Object detection with the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md]:
```suggestion
The example below demonstrates how to perform object detection with the [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/detr.md [hunk 4]
```diff
@@ -14,135 +14,132 @@ rendered properly in your Markdown viewer.
...
+
+<hfoptions id="usage">
+<hfoption id="AutoModel">
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md:54]: Missing an example with the `Pipeline`.

File: docs/source/en/model_doc/detr.md [hunk 5]
```diff
@@ -14,135 +14,132 @@ rendered properly in your Markdown viewer.
...
-```py
->>> config = DetrConfig(use_pretrained_backbone=False)
->>> model = DetrForObjectDetection(config)
+```python
+config = DetrConfig(use_pretrained_backbone=False)
+model = DetrForObjectDetection(config)
 ```
 
-As a summary, consider the following table:
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview.md) overview for more available quantization backends.
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md]: No need to include a quantization example since DETR is pretty small

File: docs/source/en/model_doc/detr.md [hunk 6]
```diff
@@ -14,135 +14,132 @@ rendered properly in your Markdown viewer.
...
+# Print detection results
+for result in results:
+    for score, label_id, box in zip(result["scores"], result["labels"], result["boxes"]):
+        score, label = score.item(), label_id.item()
+        box = [round(i, 2) for i in box.tolist()]
+        print(f"{model.config.id2label[label]}: {score:.2f} {box}")
+
+```
+
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md]: Let's keep all the `## Usage tips` in the `## Notes` section

File: docs/source/en/model_doc/detr.md [hunk 7]
```diff
@@ -14,135 +14,132 @@ rendered properly in your Markdown viewer.
...
-DETR reasons about the relations of the objects and the global image context to directly output the final set of
-predictions in parallel. The new model is conceptually simple and does not require a specialized library, unlike many
-other modern detectors. DETR demonstrates accuracy and run-time performance on par with the well-established and
-highly-optimized Faster RCNN baseline on the challenging COCO object detection dataset. Moreover, DETR can be easily
-generalized to produce panoptic segmentation in a unified manner. We show that it significantly outperforms competitive
-baselines.*
-
-This model was contributed by [nielsr](https://huggingface.co/nielsr). The original code can be found [here](https://github.com/facebookresearch/detr).
-
-## How DETR works
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md:48]: Let's also keep this but maybe in a collapsible section like this:

```md
<details>
<summary>How DETR works</summary>

...

</details>
```

**Review by stevhliu — COMMENTED** [review]
Thanks!

**Comment by arponkapuria** [comment]
@stevhliu Changes made as suggested. Let me know if anything else is needed.

File: docs/source/en/model_doc/detr.md [hunk 8]
```diff
@@ -14,38 +14,87 @@ rendered properly in your Markdown viewer.
...
-new framework, called DEtection TRansformer or DETR, are a set-based global loss that forces unique predictions via
-bipartite matching, and a transformer encoder-decoder architecture. Given a fixed small set of learned object queries,
-DETR reasons about the relations of the objects and the global image context to directly output the final set of
-predictions in parallel. The new model is conceptually simple and does not require a specialized library, unlike many
-other modern detectors. DETR demonstrates accuracy and run-time performance on par with the well-established and
-highly-optimized Faster RCNN baseline on the challenging COCO object detection dataset. Moreover, DETR can be easily
-generalized to produce panoptic segmentation in a unified manner. We show that it significantly outperforms competitive
-baselines.*
+```python
+from transformers import AutoImageProcessor, AutoModelForObjectDetection
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md:57]: The `Pipeline` example can be simplified to the below:

```py
from transformers import pipeline
import torch

pipeline = pipeline(
    "object-detection", 
    model="facebook/detr-resnet-50",
    torch_dtype=torch.float16,
    device_map=0
)

pipeline("http://images.cocodataset.org/val2017/000000039769.jpg")
```

File: docs/source/en/model_doc/detr.md [hunk 9]
```diff
@@ -120,26 +171,27 @@ required for the training to be possible, since the Hungarian matching is comput
 - The size of the images will determine the amount of memory being used, and will thus determine the `batch_size`.
   It is advised to use a batch size of 2 per GPU. See [this Github thread](https://github.com/facebookresearch/detr/issues/150) for more info.
 
-There are three ways to instantiate a DETR model (depending on what you prefer):
+There are three other ways to instantiate a DETR model (depending on what you prefer):
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md:112]: Put all 3 options in bullet points (everything in `## Notes` should be bulleted)

File: docs/source/en/model_doc/detr.md [hunk 10]
```diff
@@ -170,8 +222,6 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 - Scripts for finetuning [`DetrForObjectDetection`] with [`Trainer`] or [Accelerate](https://huggingface.co/docs/accelerate/index) can be found [here](https://github.com/huggingface/transformers/tree/main/examples/pytorch/object-detection).
 - See also: [Object detection task guide](../tasks/object_detection).
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
```
Comment (stevhliu) [inline docs/source/en/model_doc/detr.md:173]: Replace the `## Resources` section with:

```md
## Resources

- Refer to these [notebooks](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/DETR) for examples of fine-tuning [`DetrForObjectDetection`] and [`DetrForSegmentation`] on a custom dataset.
```

**Review by stevhliu — APPROVED** [review]
Thanks, just a few more changes to go!

File: docs/source/en/model_doc/detr.md (hunk 8 above)
Comment (arpon-kapuria) [reply docs/source/en/model_doc/detr.md:67]: Should I keep this `result display `code or remove it from the pipeline example?

```python
# Display results
for obj in results:
    print(f"{obj['label']} - {obj['score']:.3f} - {obj['box']}")
```
